R/core_loaders.R

```diff
@@ -0,0 +1,217 @@
+}
+
+#' Load a Parquet file
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [nanoparquet::read_parquet()]
+#'
+#' @return A data frame
+#' @noRd
+load_parquet <- function(file) {
+  check_requires("Reading Parquet files", "nanoparquet")
+  nanoparquet::read_parquet(file)
+}
+
+#' Load an `.fcs` file to `AnnData`
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to ...
+#'
+#' @return An `anndata::AnnDataR6` object
+#' @noRd
+load_fcs <- function(file, ...) {
+  cli_warn("Loading FCS files is not yet supported")
+  file
+}
+
+#' Load an `.h5mu` file to `MuData`
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to `mudata::read_h5mu()`
+#'
+#' @return A `MuData` object
+#' @noRd
+load_h5mu <- function(file, ...) {
+  # Note: this is probably not the best solution
+  check_requires("Loading MuData objects", "reticulate")
+  mudata <- reticulate::import("mudata")
+
+  mudata$read_h5mu(file, ...)
+}
+
+#' Maybe display an HTML file
+#'
+#' If interactive mode is enabled, open the file in the default browser.
+#' Otherwise just return the path to the file.
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [utils::browseURL()]
+#'
+#' @return NULL if interactive mode is enabled, the path to the file otherwise
+#' @noRd
+load_html <- function(file, ...) {
+  if (is_knitr_notebook()) {
+    lines <- readLines(file)
+    return(knitr::raw_html(paste(lines, collapse = "\n")))
+  }
+
+  if (interactive()) {
+    check_requires("Opening HTML files", "utils")
+    return(utils::browseURL(file, ...))
+  }
+
+  return(file)
+}
+
+#' Load a JSON file
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [jsonlite::fromJSON()]
+#'
+#' @return A list
+#' @noRd
+load_json <- function(file, ...) {
+  check_requires("Reading JSON files", "jsonlite")
+  jsonlite::fromJSON(file, ...)
+}
+
+#' Display an `.svg`, `.jpg`, `.png` or `.gif` in the viewer
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [...]
+#'
+#' @return NULL
+#' @noRd
+load_image <- function(file, ...) {
+  # if part of a knitr document, include the SVG
+  if (is_knitr_notebook()) {
+    ext <- tools::file_ext(file)
+
+    if (knitr::is_latex_output() && ext == "svg") {
+      check_requires("Displaying SVG images in LaTeX", "rsvg")
+      pdf_file <- tempfile(fileext = ".pdf")
+      return(rsvg::rsvg_pdf(file, pdf_file, ...))
+    }
+
+    return(knitr::include_graphics(file, ...))
+  }
+
+  if (interactive()) {
+    check_requires("Displaying images", "utils")
+    return(utils::browseURL(file, ...))
+  }
+
+  file
+}
+
+#' Load an RDS file
+#'
+#' @param file Path to the file to load
+#' @param ... Additional arguments to pass to [readRDS()]
+#'
+#' @return The object stored in the RDS file
+#' @noRd
+load_rds <- function(file, ...) {
+  readRDS(file, ...)
+}
```

Comment:
I'm not sure we need this until Python is able to save `.Rds` files

Reply:
Doesn't hurt to add this, no?

If an R user would use `rdsArtifact$load()`, would they not expect this to happen?